pandas/core/indexes/base.py

```diff
@@ -2939,8 +2934,8 @@ def _get_fill_indexer_searchsorted(self, target, method, limit=None):
         """
         if limit is not None:
             raise ValueError(
-                "limit argument for %r method only well-defined "
-                "if index and target are monotonic" % method
+                f"limit argument for {method!r} method only well-defined "
```

Comment:
we're now using repr() instead of !r, see https://github.com/pandas-dev/pandas/pull/29870#discussion_r351276079